code/modules/mining/ore_box.dm

```diff
@@ -0,0 +1,123 @@
+
+/**********************Ore box**************************/
+
+/obj/structure/ore_box
+	icon = 'icons/obj/mining.dmi'
+	icon_state = "orebox0"
+	name = "ore box"
+	desc = "A heavy box used for storing ore."
+	density = 1
+	climb_allowed = TRUE //it's a fuckin' box.
+	var/last_update = 0
+	var/list/stored_ore = list()
+
+/obj/structure/ore_box/attackby(obj/item/W as obj, mob/user as mob)
+	if (istype(W, /obj/item/stack/ore))
+		if(!user.attempt_insert_item_for_installation(W, src))
+			return
+		user.visible_message(SPAN_NOTICE("[user] drops [W] into [src]."), SPAN_NOTICE("You drop [W] into [src]."))
+	else if (istype(W, /obj/item/storage/bag/ore)) //it works differently now
+		var/obj/item/storage/bag/ore/S = W
+		if(S.isOreEmpty())
+			return
+		for(var/ore in S.stored_ore)
+			if(!stored_ore[ore])
+				stored_ore[ore] = S.stored_ore[ore]
+			else
+				stored_ore[ore] += S.stored_ore[ore]
+		S.stored_ore = list() //it'll get emptied as a result.
+		S.total_ore = 0
+		user.visible_message(SPAN_NOTICE("[user] offloads ores from [W] into [src]."), SPAN_NOTICE("You offload the ores in [W] into [src]."))
+	else if (istype(W, /obj/item/storage))
+		var/obj/item/storage/S = W
+		if(!S.contents.len)
+			return
+		S.hide_from(usr)
+		for(var/obj/item/stack/ore/O in S.contents)
+			S.remove_from_storage(O, src) //This will move the item to this item's contents
+		user.visible_message(SPAN_NOTICE("[user] offloads ores from [W] into [src]."), SPAN_NOTICE("You offload the ores in [W] into [src]."))
+
+/obj/structure/ore_box/examine(mob/user, dist)
+	. = ..()
+	if(isEmpty())
+		. += SPAN_NOTICE("It is empty.")
+		return
+	. += SPAN_NOTICE("It holds:")
+	for(var/ore in stored_ore)
+		var/obj/item/stack/ore/O = ore
+		. += SPAN_NOTICE("- [stored_ore[ore]] [initial(O.name)]")
+
+/// Sigh.
+/obj/structure/ore_box/Entered(atom/movable/AM, atom/oldLoc)
+	. = ..()
+	if(istype(AM, /obj/item/stack/ore))
+		take(AM)
+
+/obj/structure/ore_box/drop_products(method, atom/where)
+	. = ..()
+	var/i = 0
+	while(!isEmpty() && i++ < 200) //may be laggy as hell if they're carrying 2,000+ sand, so capping it at 200 items dropped.
```

Comment:
It's minor, but incrementing `i` in the loop body instead of the control statement would be clearer.